Sources/NIO/AddressedEnvelope.swift

```diff
@@ -20,15 +20,35 @@
 public struct AddressedEnvelope<DataType> {
     public var remoteAddress: SocketAddress
     public var data: DataType
+    /// Any metadata associated with this `AddressedEnvelope`
+    public var metadata: Metadata? = nil
 
     public init(remoteAddress: SocketAddress, data: DataType) {
         self.remoteAddress = remoteAddress
         self.data = data
     }
+    
+    /// Any metadata associated with an `AddressedEnvelope`
+    public struct Metadata {
```

Comment:
Let's make this struct `Hashable` too, while we're here.

Reply:
Done.